### PR TITLE
docs: show old documentation warning in Vitest 1 docs 

### PR DESCRIPTION
**Pull request by sheremet-va**
### Description

See a similar issue in https://github.com/vitejs/vite/pull/18805

I will add the same waring to pre-stable Vitest 0 after we approve the design

Screenshot 2024-12-02 at 11 10 37
Screenshot 2024-12-02 at 11 38 48

Related https://github.com/vitest-dev/vitest/issues/6980

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: docs/.vitepress/components/OldDocument.vue [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  top: 0;
+  left: 0;
+  display: flex;
+  height: var(--vp-layout-top-height);
+  width: 100%;
+  padding: 4px 32px;
+  justify-content: center;
+  align-items: center;
+  color: var(--vp-c-text-1);
+  background: var(--vp-c-brand-softest);
```
Comment (AriPerkkio) [inline docs/.vitepress/components/OldDocument.vue:23]: How about making this more visible?

```suggestion
  background-color: var(--vp-c-caution-3);
```

File: packages/coverage-istanbul/src/provider.ts [hunk 2]
```diff
@@ -13,7 +13,8 @@ import libCoverage from 'istanbul-lib-coverage'
 import libSourceMaps from 'istanbul-lib-source-maps'
 import { type Instrumenter, createInstrumenter } from 'istanbul-lib-instrument'
 
-// @ts-expect-error missing types
+// eslint-disable-next-line ts/prefer-ts-expect-error
+// @ts-ignore missing types
```
Comment (AriPerkkio) [inline packages/coverage-istanbul/src/provider.ts:17]: These ignore comments are no longer needed as types are present. Maybe rebase from `main` helps.

https://github.com/vitest-dev/vitest/blob/e2c29eaf0158aa7c9b6d6e8e750fd24f70e65420/packages/coverage-istanbul/src/provider.ts#L10-L12

**Review by AriPerkkio — APPROVED** [review]
This is great addition 👍

File: packages/coverage-istanbul/src/provider.ts (hunk 2 above)
Comment (sheremet-va) [reply packages/coverage-istanbul/src/provider.ts:17]: This is v1, we didn't have types back then 🤔

Comment (AriPerkkio) [reply packages/coverage-istanbul/src/provider.ts:17]: Ahh okey I see. Feel free to change the lines as did 😄

**Review by hi-ogawa — APPROVED** [review]
Looks good!

**Comment by sheremet-va** [comment]
Closed in favor of https://github.com/vitest-dev/vitest/pull/7003

**Pull request closed without merging**